### PR TITLE
encapp: move get_app_id to encapp_tool package and rename functions

### PR DESCRIPTION
**Pull request by hidekimendoza**
Move get_app_id function from encapp.py to encapp_tool.adb_cmds,
unittest added.
Add encapp to function names in encapp.py if APPNAME_MAIN is used

### CODE REVIEW

File: scripts/encapp.py [hunk 1]
```diff
@@ -180,14 +170,14 @@ def grant_camera_permission(serial, debug):
...
-def install_ok(serial, debug=0):
+def is_encapp_installed(serial, debug=0):
     package_list = installed_apps(serial, debug)
     if APPNAME_MAIN not in package_list:
         return False
     return True
 
 
-def uninstall_app(serial, debug=0):
+def uninstall_encapp_app(serial, debug=0):
```
Comment (chemag) [inline scripts/encapp.py:180]: I wouldn't do this: the encapp part is already in the module name

File: scripts/encapp.py [hunk 2]
```diff
@@ -180,14 +170,14 @@ def grant_camera_permission(serial, debug):
             'android.permission.CAMERA', debug)
 
 
-def install_ok(serial, debug=0):
+def is_encapp_installed(serial, debug=0):
```
Comment (chemag) [inline scripts/encapp.py:173]: I wouldn't do this: the encapp part is already in the module name

File: scripts/encapp.py [hunk 3]
```diff
@@ -135,28 +135,18 @@ def remove_encapp_gen_files(serial, debug=0):
     remove_files_using_regex(serial, regex_str, location, debug)
 
 
-def wait_for_exit(serial, debug=0):
-    adb_cmd = f'adb -s {serial} shell pidof {APPNAME_MAIN}'
+def wait_for_encapp_exit(serial, debug=0):
```
Comment (chemag) [inline scripts/encapp.py:138]: I wouldn't do this: the encapp part is already in the module name

File: scripts/encapp_tool/adb_cmds.py [hunk 4]
```diff
@@ -20,9 +20,9 @@ def run_cmd(cmd: str, debug: int = 0) -> Tuple[bool, str, str]:
     try:
         if debug > 0:
             print(cmd, sep=" ")
-        process = Popen(cmd, shell=True, stdout=PIPE, stderr=PIPE)  # noqa: P204
-        stdout, stderr = process.communicate()
-        ret = True if process.returncode == 0 else False
+        with Popen(cmd, shell=True, stdout=PIPE, stderr=PIPE) as process:
```
Comment (chemag) [inline scripts/encapp_tool/adb_cmds.py:23]: nice. 

Does this mean that we're leaking if we don't do close the process?

File: scripts/encapp_tool/adb_cmds.py [hunk 5]
```diff
@@ -55,13 +55,16 @@ def get_device_info(serial_inp: Optional[str], debug=0) -> Tuple[Dict, str]:
     if serial_inp is None:
         # if user did not select a serial_inp, make sure there is only one
         # device available
-        assert len(device_info) == 1, f"error: need to choose a device [{', '.join(device_info.keys())}]"
+        assert (
```
Comment (chemag) [inline scripts/encapp_tool/adb_cmds.py:58]: suggest to move all lint changes to a separate patch (it makes it easier for reviewers)

File: scripts/encapp_tool/adb_cmds.py [hunk 6]
```diff
@@ -124,3 +127,27 @@ def get_connected_devices(debug: int) -> Dict:
             item_dict["model"] = "generic"
         device_info[serial] = item_dict
     return device_info
+
+
+def get_app_pid(serial: str, package_name: str, debug=0):
```
Comment (chemag) [inline scripts/encapp_tool/adb_cmds.py:132]: nice

File: scripts/tests/adb_cmds_unittest.py [hunk 7]
```diff
@@ -73,9 +73,7 @@ def test_get_device_info_with_multiple_device_not_specified_return_info(
         self, mock_run
     ):
         mock_run.return_value = (True, ADB_DEVICES_MULTI_DEV, "")
-        error_str = (
-            "error: need to choose a device [1234567890abcde, abcde1234567890]"
-        )
+        error_str = "error: need to choose a device [1234567890abcde, abcde1234567890]"
```
Comment (chemag) [inline scripts/tests/adb_cmds_unittest.py:76]: move lint to a different patch

File: scripts/tests/adb_cmds_unittest.py [hunk 8]
```diff
@@ -116,6 +114,24 @@ def test_remove_files_using_regex_shall_remove_regex_match_files(self, mock_run)
             any_order=True,
         )
 
+    @patch("encapp_tool.adb_cmds.run_cmd")
```
Comment (chemag) [inline scripts/tests/adb_cmds_unittest.py:117]: what does this do?

**Review by chemag — CHANGES_REQUESTED** [review]

File: scripts/tests/adb_cmds_unittest.py (hunk 8 above)
Comment (hidekimendoza) [reply scripts/tests/adb_cmds_unittest.py:119]: It mocks the run_cmd for unit testing

File: scripts/encapp_tool/adb_cmds.py (hunk 4 above)
Comment (JohanBlome) [reply scripts/encapp_tool/adb_cmds.py:23]: I may if there is failure, https://bugs.python.org/issue43308.

**Review by JohanBlome — APPROVED** [review]
